### PR TITLE
Correct typing of CypressRunResult::runs

### LINKED ISSUE #6126: cypress-cli type for CypressRunResult::runs is incorrect (can be null/undefined)
### Current behavior:

In the event that the browser fails to connect, the run promise resolves with results which has `runs` of `null` or `undefined`. (Which is likely a separate issue altogether. It should probably not resolve at all - it should reject).

However, the typings indicate that the type of `runs` is `RunResult[]` which doesn't match `null`

### Desired behavior:

The type should indicate that `runs` can be undefined or null, or `run()` should not resolve on an error running the tests altogether.

### Test code to reproduce

1. Clone https://github.com/xdumaine/cypress-test-tiny
2. `npm i && npm test`

### PR DESCRIPTION
**Pull request by xdumaine**
CypressRunResult::runs can be null if the browser fails to connect

Short term fix for #6126 but the long term fix is to change how `cypress.run()` resolves/rejects based on failing to start the browser.

- Closes #6126

### User facing changelog

Changes the type definition for CypressRunResult::runs to reflect that it can be undefined or null.

### How has the user experience changed?

Builds won't explode because code can be correctly typed and falsy states handled.

### PR Tasks

None of the predefined ones applied, please feel free to add some if necessary.

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
Hey @xdumaine, will you be able to address the changes asked of for this PR above by bahmutov? If not, we will need to close this due to inactivity.

**Comment by xdumaine** [comment]
@jennifer-shehane Yes, thanks for the poke, I should be able to address this week.

**Comment by xdumaine** [comment]
That build failure seems unrelated. If not, could you point me toward what's going on?

**Comment by xdumaine** [comment]
@jennifer-shehane @bahmutov  I addressed the requested changes here

**Comment by CypressJoseph** [comment]
The code changes look good to me, @xdumaine! Great work. However, it does look like we lost the additional JSDoc hints around the attributes of the FailedRun type (as well as the example of using the shape.) Would you be able to resurrect/reconstruct the docs for this interface?

File: cli/types/cypress-npm-api.d.ts [hunk 1]
```diff
@@ -475,29 +475,16 @@ declare module 'cypress' {
...
-      ```
-      const result = await cypress.run()
-      if (result.failures) {
-        console.error(result.message)
-        process.exit(result.failures)
-      }
-      ```
-     */
-    failures?: number
-    /**
```
Comment (CypressJoseph) [inline cli/types/cypress-npm-api.d.ts]: Can we reconstruct these attribute docs for the new shape?

**Comment by jennifer-shehane** [comment]
@xdumaine Do you still have time to address the changes requested in this PR? We will have to close this PR due to inactivity otherwise.

**Comment by xdumaine** [comment]
Just close it if you don't want it. I provided a fix to a bug, but if you
don't want it, that's on you. I'm not going to go back and forth repeatedly
on such a small change. The reviewer can make the doc change if he wants
to. As is, the documentation reflects the types accurately as far as I can
tell.

**Comment by CypressJoseph** [comment]
@xdumaine I have reconstructed the example snippet which describes how to introspect on the run failure (it does seem important to me that we don't lose the guidance to the user about how to work with this shape)

**Pull request merged**